src/unstructured_client/users.py

```diff
@@ -458,3 +467,125 @@ async def store_secret_async(
+            padding.OAEP(
+                mgf=padding.MGF1(algorithm=hashes.SHA256()),
+                algorithm=hashes.SHA256(),
+                label=None
+            )
+        )
+
+        return {
+            'encrypted_aes_key': base64.b64encode(encrypted_key).decode('utf-8'),
+            'aes_iv': base64.b64encode(iv).decode('utf-8'),
+            'encrypted_value': base64.b64encode(ciphertext).decode('utf-8'),
+            'type': 'rsa_aes',
+        }
+
+    def _encrypt_rsa(
+            self,
+            encryption_key_pem: str,
+            plaintext: str,
+    ) -> dict:
+        # Load public RSA key
+        public_key = serialization.load_pem_public_key(
+            encryption_key_pem.encode('utf-8'),
+            backend=default_backend()
+        )
+
+        if not isinstance(public_key, rsa.RSAPublicKey):
+            raise TypeError("Public key must be an RSA public key for encryption.")
+
+        ciphertext = public_key.encrypt(
+            plaintext.encode(),
+            padding.OAEP(
+                mgf=padding.MGF1(algorithm=hashes.SHA256()),
+                algorithm=hashes.SHA256(),
+                label=None
+            ),
+        )
+        return {
+            'encrypted_value': base64.b64encode(ciphertext).decode('utf-8'),
+            'type': 'rsa',
+            'encrypted_aes_key': "",
+            'aes_iv': "",
+        }
+
+
+    def encrypt_secret(
+            self,
+            encryption_cert_or_key_pem: str,
+            plaintext: str,
+            encryption_type: Optional[str] = None,
+    ) -> dict:
+        """
+        Encrypts a plaintext string for securely sending to the Unstructured API.
+        
+        Args:
+            encryption_cert_or_key_pem (str): A PEM-encoded RSA public key or certificate.
+            plaintext (str): The string to encrypt.
+            type (str, optional): Encryption type, either "rsa" or "rsa_aes".
+
+        Returns:
+            dict: A dictionary with encrypted AES key, iv, and ciphertext (all base64-encoded).
+        """
+        # If a cert is provided, extract the public key
+        if "BEGIN CERTIFICATE" in encryption_cert_or_key_pem:
+            cert = x509.load_pem_x509_certificate(
+                encryption_cert_or_key_pem.encode('utf-8'),
+            )
+
+            loaded_key = cert.public_key()
+
+            # Serialize back to PEM format for consistency
+            public_key_pem = loaded_key.public_bytes(
+                encoding=serialization.Encoding.PEM,
+                format=serialization.PublicFormat.SubjectPublicKeyInfo
+            ).decode('utf-8')
+
+        else:
+            public_key_pem = encryption_cert_or_key_pem
+
+        # If the plaintext is short, use RSA directly
+        # Otherwise, use a RSA_AES envelope hybrid
+        # The length of the public key is a good hueristic
+        if not encryption_type:
+            encryption_type = "rsa" if len(plaintext) <= len(public_key_pem) else "rsa_aes"
```

Comment:
Can we load the key in this function and then check the size like
```python
    max_payload = (public_key.key_size // 8) - 42  # OAEP overhead
    plaintext_bytes = len(plaintext.encode('utf-8'))
    encryption_type = "rsa" if plaintext_bytes <= max_payload else "rsa_aes"
```

I think this would fail when in some cases. For example
1. Using RSA 2048. Max Payload is ((2048/8 bits) - 42 byte OAEP) = 214 bytes
2. 215 byte plaintext. 215 < 451 PEM which means RSA is selected
3. Attempt to encrypt 215 byte plain text but max payload is 214